crates/store/re_log_encoding/src/rrd/footer/instances.rs

```diff
@@ -13,4 +28,807 @@
+            }
+    }
+}
+
+// Helpers
+impl RrdManifest {
+    pub fn compute_sorbet_schema_sha256(
+        schema: &arrow::datatypes::Schema,
+    ) -> Result<[u8; 32], arrow::error::ArrowError> {
+        let schema = {
+            // Sort and remove top-level metadata before hashing.
+            let mut fields = schema.fields().to_vec();
+            fields.sort();
+            arrow::datatypes::Schema::new_with_metadata(fields, Default::default()) // no metadata!
+        };
+
+        let partition_schema_ipc = {
+            let mut schema_ipc = Vec::new();
+            arrow::ipc::writer::StreamWriter::try_new(&mut schema_ipc, &schema)?;
+            schema_ipc
+        };
+
+        use sha2::Digest as _;
+        let mut hash = [0u8; 32];
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(&partition_schema_ipc);
+        hasher.finalize_into(sha2::digest::generic_array::GenericArray::from_mut_slice(
+            &mut hash,
+        ));
+
+        Ok(hash)
+    }
+
+    /// Computes the appropriate column name for the provided parts.
+    ///
+    /// The name is guaranteed to be sanitized and safe to use in all environments where Rerun data
+    /// can generally be found (Lance, external dataframe libraries, etc).
+    ///
+    /// If caller doesn't provide any part (i.e. all are `None`), an empty string is returned.
+    pub fn compute_column_name(
+        entity_path: Option<&EntityPath>,
+        strip_entity_prefix: Option<&str>,
+        component_desc: Option<&ComponentDescriptor>,
+        prefix: Option<&str>,
+        suffix: Option<&str>,
+    ) -> String {
+        use re_types_core::reflection::ComponentDescriptorExt as _;
+        let full_name = [
+            prefix.map(ToOwned::to_owned),
+            entity_path.map(|p| {
+                let path = p.to_string();
+                // Optionally strip the entity prefix if provided
+                let path = strip_entity_prefix
+                    .and_then(|prefix| path.strip_prefix(prefix))
+                    .unwrap_or(&path);
+                // Always strip trailing slashes (if present)
+                path.strip_suffix("/").unwrap_or(path).to_owned()
+            }),
+            component_desc
+                .and_then(|descr| descr.archetype)
+                .map(|archetype| archetype.short_name().to_owned()),
+            component_desc.map(|descr| descr.archetype_field_name().to_owned()),
+            suffix.map(ToOwned::to_owned),
+        ]
+        .into_iter()
+        .flatten()
+        .filter(|s| !s.is_empty())
+        .collect::<Vec<_>>()
+        .join(":");
+
+        // All of the following characters have proven to be problematic in one or more
+        // environments. Replace them with `_`, unconditionally.
+        let sanitized = full_name.replace([',', ' ', '-', '.', '\\'], "_");
+
+        // Remove leading underscore if present
+        sanitized.trim_start_matches('_').to_owned()
+    }
+}
+
+// Sanity checks
```

Comment:
RRD manifests are effectively user input so we need to be at least a little on the defensive so we don't register complete non-sense. All the checks marked `cheap` basically only look at schema concerns, never the data itself.

There are some extra paranoid layers on the dataplatform side.